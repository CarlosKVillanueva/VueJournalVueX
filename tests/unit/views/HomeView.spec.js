import { shallowMount } from "@vue/test-utils"
import HomeView from "@/views/HomeView";


describe( 'Home View Test', () => {

    test( 'Must render the snapshot succesfully', () => {

        const wrapper = shallowMount( HomeView )

        expect( wrapper.html() ).toMatchSnapshot()
    } );
    
    test('On Button click, must redirect to "no-entry" ', () => {
        const mockRouter = {
            push: jest.fn()
        }

        const wrapper = shallowMount( HomeView, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        } )
        wrapper.find( 'button' ).trigger( 'click' )

        expect( mockRouter.push ).toHaveBeenCalledWith({name: 'no-entry'})
    });
} );