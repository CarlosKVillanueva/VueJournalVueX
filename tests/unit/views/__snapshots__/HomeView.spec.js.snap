// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home View Test Must render the snapshot succesfully 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hola Mundo</h1><button class="btn btn-primary">Go to DayBook</button>
</div>
`;
